src/vs/workbench/parts/terminal/electron-browser/terminalPanel.ts

```diff
@@ -370,4 +376,36 @@ export class TerminalPanel extends Panel {
 		}
 		return path;
 	}
+
+	/**
+	 * Surrounds a Windows path with quotation marks if necessary
+	 */
+	public static prepareWindowsPathForTerminal(path: string): string {
+		if (/\s+/.test(path)) {
+			return `"${path}"`;
+		}
+		return path;
+	}
+
+	/**
+	 * Adds quotes to a path if it contains whitespaces, and formats it for the correct windows subshell.
+	 * It is assumed there is an active terminal instance.
+	 */
+	private _preparePathForTerminal(path: string): string {
+		if (platform.isWindows) {
+			const terminal = this._terminalService.getActiveInstance();
+			const shell = terminal.getShellName();
+			for (const [format, formatter, isWinPath] of WIN_PATH_FORMATTERS) {
```

Comment:
Okay.

Is the following pattern used at all?

```typescript
WIN_PATH_FORMATTERS.forEach(formatArray => {
    const [format, formatter, isWinPath] = formatArray;
    ...
}
```
Or if I switched to interfaces I wouldn't have to worry about this.